source/dynamical_systems/tests/test_circular.cpp

```diff
@@ -0,0 +1,33 @@
+#include "dynamical_systems/Circular.hpp"
+#include <gtest/gtest.h>
+#include <unistd.h>
+
+using namespace std::literals::chrono_literals;
+
+class CircularDSTest : public testing::Test {
+protected:
+  void SetUp() override {
+    current_pose = StateRepresentation::CartesianPose("A", 10 * Eigen::Vector3d::Random());
+    center = StateRepresentation::CartesianPose::Identity("B");
+  }
+
+  StateRepresentation::CartesianPose current_pose;
+  StateRepresentation::CartesianPose center;
+  double radius = 10;
+  unsigned int nb_steps = 2000;
+  std::chrono::milliseconds dt = 10ms;
+  double linear_tol = 1e-3;
+};
+
+TEST_F(CircularDSTest, TestPositionOnRadius) {
+  DynamicalSystems::Circular circularDS(center);
```

Comment:
Additionally, I have a question about the circular DS (that is not clear to me, and should maybe be more explicit in the code): So we initialize the circular DS with a radius, center, and gains (did I forget something?). Now, how do we know in which sense the current_pose will turn around the center? Is it the right hand rule? And if I wanted the DS to turn in the opposite direction, could I just rotate the center PI around its x or y axis?

Reply:
If I remember there is another parameter that is the circular velocity. Putting it negative changes the rotation direction.